api/src/main/java/io/opentelemetry/trace/SpanContext.java

```diff
@@ -75,7 +75,7 @@ public static SpanContext create(
    * @since 0.1.0
    */
   public static SpanContext createFromRemoteParent(
```

Comment:
To avoid allocations it is probably better to have the ctor that allows to pass a charsequnce and a position from where it starts. 

CharSequnce traceIdBuf, int traceIdOffset for the current ctor all the offsets will be 0 but for w3c we can avoid calling substring and internally we can still store longs if we want or we can for the moment do the substring.

Similar ctor can be offered for bytes Buffer (not that class probably a byte array as a buffer).